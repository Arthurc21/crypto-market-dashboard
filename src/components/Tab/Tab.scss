.tab {
    border-top-left-radius: 6px;
    border-top-right-radius: 6px;
    height: fit-content;
    width: fit-content;
    background-color: #D0D9DD;

    .span-wrapper {
        font-weight: 600;
        font-size: 20px;
        margin: 12px 20px;
        color: #FFFFFF;
    }
}