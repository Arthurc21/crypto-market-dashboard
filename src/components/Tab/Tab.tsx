import React from 'react'
import './Tab.scss'

interface TabProps {
	text: string
	value: string
	onClick(value: string): void
	selected: boolean
}

export const Tab: React.FC<TabProps> = ({ text, value, onClick, selected = false }: TabProps) => {
	return (
		<div className={`tab ${selected ? 'selected' : ''}`} onClick={() => onClick(value)}>
			<div className="span-wrapper">
				<span>{text}</span>
			</div>
		</div>
	)
}
