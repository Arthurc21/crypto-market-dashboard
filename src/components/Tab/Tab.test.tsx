import { create } from 'react-test-renderer'
import { Tab } from './Tab'

describe('Tab Testing', () => {
	const onClickFunc = jest.fn()
	test('tab snapshot', () => {
		let tab = create(
			<Tab
				text={'tabText'}
                value={'tabTestValue'}
                onClick={onClickFunc}
                selected={true}
			/>
		)
		expect(tab.toJSON()).toMatchSnapshot()
	})
})