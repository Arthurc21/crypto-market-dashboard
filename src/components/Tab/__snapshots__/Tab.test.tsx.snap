// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Testing tab snapshot 1`] = `
<div
  className="tab selected"
  onClick={[Function]}
>
  <div
    className="span-wrapper selected"
  >
    <span>
      tabText
    </span>
  </div>
</div>
`;
